common/countdown/countdown.go

```diff
@@ -15,7 +15,7 @@ type CountdownTimer struct {
 	initilised      bool
 	timeoutDuration time.Duration
 	// Triggered when the countdown timer timeout for the `timeoutDuration` period, it will pass current timestamp to the callback function
-	OnTimeoutFn func(time time.Time) error
+	OnTimeoutFn func(time time.Time, i interface{}) error
```

Comment:
I guess `i` is chainReader. I am not sure if this is better:
```
type CountdownTimer struct {
    ....
    chain *chainReader
}
```

Reply:
no, otherwise you are coupling chain with timer